docs/api/cli.md

```diff
@@ -0,0 +1,105 @@
+# Command line utility (fda)
+
+`fda` is a command line utility for interacting with the Feldera Manager's REST API.
+
+## Installation
+
+In order to install `fda`, you need a working Rust environment. You can install Rust by following the instructions on
+the [Rust website](https://www.rust-lang.org/tools/install).
+
+### Using Cargo
+
+Install `fda` with Cargo by running the following command:
+
+```commandline
+cargo install fda
+```
+
+Alternatively, to install the latest `fda` revision from our main git branch, run the following command:
+
+```commandline
+cargo install --git https://github.com/feldera/feldera fda
+```
+
+To install from the sources in your local feldera repository, you can install `fda` with the
+following commands:
+
+```commandline
+cd crates/feldera-cli
+cargo install --path .
+```
+
+### Optional: Shell completion
+
+Once the `fda` binary is installed, you can enable shell command completion for `fda` by generating a completion script
+for your shell and copying it to the appropriate location.
+
+```commandline
+fda shell-completion --help
+fda shell-completion
+```
+
+Currently, Bash, Elvish, Fish, Powershell, and Zsh are supported.
+
+## Connecting & Authentication
+
+To connect to the Feldera manager, you need to provide the URL of the manager. You can either provide the URL as an
+environment variable or as a command line argument. The environment variable is called `FELDERA_HOST` and the
+command line argument is called `--host`.
+
+If your Feldera instance requires authentication (not needed in the local docker form factor), you'll also need to 
+provide an API key. You can either set the API key as an environment variable or as a command line argument.
+The environment variable is called `FELDERA_API_KEY` and the command line argument is called `--auth`.
+It is recommended to use an environment variable configured in your shell init script to avoid storing the API
+key in your shell history.
+
+:::info
+
+You can create a new API key by logging into the Feldera WebConsole. Once logged in, click on your profile in the top
+right. Go to **Manage API Keys** and click **Generate new key**.
+
+:::
+
+## Usage Examples
```

Comment:
What's the main motivation to have it be `command object-type object-name object-action` rather than `command object-action object-type object-name`?

E.g.,
- `fda pipeline p1 start` would become `fda start pipeline p1`
- `fda pipeline p1 program` would become `fda get pipeline p1 --program-only`

It might just depend on the user base familiarity which one is preferred?

Reply:
(Or `fctl`)

Reply:
I felt it was simpler to learn for a user as the first thing they'll likely think about is the entity they want to work on pipeline (or api-key or whatever else gets added in the future), then they can your way down from there (and autocomplete gives you the list of things only relevant to pipeline now). But mostly just a choice, we can revise in the future if it's a problem.